adapters/telaria/telaria.go

```diff
@@ -0,0 +1,331 @@
+func (a *TelariaAdapter) FetchTelariaExtImpParams(imp *openrtb.Imp) (*openrtb_ext.ExtImpTelaria, error) {
+	var bidderExt adapters.ExtImpBidder
+	err := json.Unmarshal(imp.Ext, &bidderExt)
+
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+
+		return nil, err
+	}
+
+	var telariaExt openrtb_ext.ExtImpTelaria
+	_ = json.Unmarshal(bidderExt.Bidder, &telariaExt)
+
+	return &telariaExt, nil
+}
+
+// Fetch the id from the appropriate publisher object
+func (a *TelariaAdapter) FetchPublisherId(request *openrtb.BidRequest) string {
+
+	if request.Site != nil {
+		if request.Site.Publisher != nil {
+			if request.Site.Publisher.ID != "" {
+				return request.Site.Publisher.ID
+			}
+		}
+	}
+
+	if request.App != nil {
+		if request.App.Publisher != nil {
+			if request.App.Publisher.ID != "" {
+				return request.App.Publisher.ID
+			}
+		}
+	}
+
+	return ""
+}
+
+// This method changes <site/app>.publisher.id to request.ext.telaria.seatCode
+// And moves the publisher.id to request.ext.originalPublisherId
+func (a *TelariaAdapter) PopulateRequestExtAndPubId(request *openrtb.BidRequest) error {
+	var requestExtIncoming ReqExtIn
+	if err := json.Unmarshal(request.Ext, &requestExtIncoming); err != nil {
+		return err
+	}
+
+	if requestExtIncoming.Telaria.SeatCode == "" {
+		return &errortypes.BadInput{Message: "Seat Code is required"}
+	}
+
+	request.Ext, _ = json.Marshal(&BidExtOut{requestExtIncoming.ExtRequest, a.FetchPublisherId(request)})
+
+	publisherObject := &openrtb.Publisher{ID: requestExtIncoming.Telaria.SeatCode}
+
+	if request.Site != nil {
+		if request.Site.Publisher != nil {
+			request.Site.Publisher.ID = requestExtIncoming.Telaria.SeatCode
+		} else {
+			request.Site.Publisher = publisherObject
+		}
+	}
+
+	if request.App != nil {
+		if request.App.Publisher != nil {
+			request.App.Publisher.ID = requestExtIncoming.Telaria.SeatCode
+		} else {
+			request.App.Publisher = publisherObject
+		}
+	}
+
+	return nil
+}
+
+func (a *TelariaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	if noImps := a.CheckHasImps(request); noImps != nil {
+		return nil, []error{noImps}
+	}
+
+	if noVideoObjectError := a.CheckHasVideoObject(request); noVideoObjectError != nil {
+		return nil, []error{noVideoObjectError}
+	}
+
+	if noSeatCodeError := a.PopulateRequestExtAndPubId(request); noSeatCodeError != nil {
+		return nil, []error{noSeatCodeError}
+	}
+
+	var errors []error
+	for i, imp := range request.Imp {
+		telariaExt, err := a.FetchTelariaExtImpParams(&imp)
+
+		if err != nil {
+			errors = append(errors, err)
+		} else {
+			request.Imp[i].TagID = telariaExt.AdCode
+			request.Imp[i].Ext, _ = json.Marshal(&ImpressionExtOut{request.Imp[i].TagID})
+		}
+	}
+
+	if len(errors) > 0 {
+		return nil, errors
+	}
+
+	reqJSON, _ := json.Marshal(request)
```

Comment:
Again, are we sure no `json.Marshal` error could get returned under every circumstance?

Reply:
I will add an err check at these places.